docs/csharp/language-reference/builtin-types/record.md

```diff
@@ -0,0 +1,205 @@
+
+The derived record declares positional parameters for all the parameters in the base record primary constructor. The base record declares and initializes those properties. The derived record doesn't hide them, but only creates and initializes properties for parameters that aren't declared in its base record.
+
+The following example illustrates inheritance with positional property syntax:
+
+:::code language="csharp" source="snippets/shared/RecordType.cs" id="PositionalInheritance":::
+
+### Equality in inheritance hierarchies
+
+For two record variables to be equal, the run-time type must be equal. The type of the containing variable might be different. This is illustrated in the following code example:
+
+:::code language="csharp" source="snippets/shared/RecordType.cs" id="InheritanceEquality":::
+
+In the example, all instances have the same properties and the same property values. But `student == teacher` returns `False` although both are `Person`-type variables, and `student == student2` returns `True` although one is a `Person` variable and one is a `Student` variable.
+
+To implement this behavior, the compiler synthesizes an `EqualityContract` property that returns a <xref:System.Type> object that matches the type of the record. This enables the equality methods to compare the runtime type of objects when they are checking for equality. If the base type of a record is `object`, this property is `virtual`. If the base type is another record type, this property is an override. If the record type is `sealed`, this property is `sealed`.
+
+When comparing two instances of a derived type, the synthesized equality methods check all properties of the base and derived types for equality. The synthesized `GetHashCode` method uses the `GetHashCode` method from all properties and fields declared in the base type and the derived record type.
+
+### `with` expressions in derived records
+
+Because the synthesized clone method uses a [covariant return type](~/_csharplang/proposals/csharp-9.0/covariant-returns.md), the result of a `with` expression has the same run-time type as the expression's operand. All properties of the run-time type get copied, but you can only set properties of the compile-time type, as the following example shows:
+
+:::code language="csharp" source="snippets/shared/RecordType.cs" id="WithExpressionInheritance":::
+
+### `PrintMembers` formatting in derived records
+
+the synthesized `PrintMembers` method of a derived record type calls the base implementation. The result is that all public properties and fields of both derived and base types are included in the `ToString` output, as shown in the following example:
+
+:::code language="csharp" source="snippets/shared/RecordType.cs" id="ToStringInheritance":::
+
+You can provide your own implementation of the `PrintMembers` method. If you do that, use the following signature:
+
+* For a `sealed` record: `private bool PrintMembers(StringBuilder builder)`;
```

Comment:
This needs to be clarified a bit:

```suggestion
* For a `sealed` record that derives from `object`: `private bool PrintMembers(StringBuilder builder)`;
```

IIRC, a sealed record that derives from another record would have the signature `protected sealed override bool PrintMembers(StringBuilder builder);`